Astro V4 compatible

Dependency incompatibility error when running `npm install` with astro v4

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: lucide-astro@0.294.0
npm ERR! Found: astro@4.0.3
npm ERR! node_modules/astro
npm ERR!   astro@"^4.0.1" from the root project
npm ERR!   peer astro@"^4.0.0-beta.0" from @astrojs/markdown-remark@4.0.0
npm ERR!   node_modules/@astrojs/markdown-remark
npm ERR!     @astrojs/markdown-remark@"4.0.0" from @astrojs/mdx@2.0.0
npm ERR!     node_modules/@astrojs/mdx
npm ERR!       @astrojs/mdx@"^2.0.0" from @astrojs/starlight@0.15.0
npm ERR!       node_modules/@astrojs/starlight
npm ERR!         @astrojs/starlight@"^0.15.0" from the root project
npm ERR!     @astrojs/markdown-remark@"4.0.0" from astro@4.0.3
npm ERR!   5 more (@astrojs/mdx, @astrojs/starlight, @astrojs/svelte, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer astro@"^2.7.1 || ^3.0.0" from lucide-astro@0.294.0
npm ERR! node_modules/lucide-astro
npm ERR!   lucide-astro@"^0.294.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: astro@3.6.4
npm ERR! node_modules/astro
npm ERR!   peer astro@"^2.7.1 || ^3.0.0" from lucide-astro@0.294.0
npm ERR!   node_modules/lucide-astro
npm ERR!     lucide-astro@"^0.294.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/jack/.npm/_logs/2023-12-11T14_18_44_416Z-eresolve-report.txt
```

--- a/.editorconfig
+++ b/.editorconfig
@@ -1,0 +1,12 @@
+# EditorConfig is awesome: https://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+[*]
+indent_style = tab
+indent_size = 4
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true

--- a/.github/workflows/build_publish.yml
+++ b/.github/workflows/build_publish.yml
@@ -18,31 +18,41 @@ jobs:
         with:
           persist-credentials: false 
           fetch-depth: 0
+
       - uses: actions/setup-node@v3
         with:
           node-version: 20
           registry-url: https://registry.npmjs.org/
+
+      - name: Prepare Upgrade script
+        run: npm ci
+
       - name: Upgrade
         id: upgrade
         run: node upgrade.js
         continue-on-error: true
+
       - name: Install
         run: npm i
         if: steps.upgrade.outcome == 'success' && steps.upgrade.conclusion == 'success'
+
       - name: Build
         run: npm run build
         if: steps.upgrade.outcome == 'success' && steps.upgrade.conclusion == 'success'
+
       - name: Publish on NPM
         run: npm publish
         if: steps.upgrade.outcome == 'success' && steps.upgrade.conclusion == 'success'
         env:
           NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+
       - name: Commit files
         run: |
           git config --local user.email "github@dzeio.com"
           git config --local user.name "Dzeio BOT"
           git commit -a -m "bump from remote"
         if: steps.upgrade.outcome == 'success' && steps.upgrade.conclusion == 'success'
+
       - name: Push changes
         uses: ad-m/github-push-action@master
         with:

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -13,16 +13,20 @@ jobs:
     steps:
       - uses: actions/checkout@v3
         with:
-          persist-credentials: false 
+          persist-credentials: false
           fetch-depth: 0
+
       - uses: actions/setup-node@v3
         with:
           node-version: 20
           registry-url: https://registry.npmjs.org/
+
       - name: Install
         run: npm i
+
       - name: Build
         run: npm run build
+
       - name: Publish on NPM
         run: npm publish
         env:

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -1,0 +1,30 @@
+
+# ENV needed:
+# GIT_TOKEN
+# NPM_TOKEN
+name: Build & Package
+
+on:
+  workflow_dispatch:
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false 
+          fetch-depth: 0
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20
+          registry-url: https://registry.npmjs.org/
+        continue-on-error: true
+      - name: Install
+        run: npm i
+      - name: Build
+        run: npm run build
+      - name: Publish on NPM
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -2,7 +2,7 @@
 # ENV needed:
 # GIT_TOKEN
 # NPM_TOKEN
-name: Build & Package
+name: Manual publish
 
 on:
   workflow_dispatch:
@@ -19,7 +19,6 @@ jobs:
         with:
           node-version: 20
           registry-url: https://registry.npmjs.org/
-        continue-on-error: true
       - name: Install
         run: npm i
       - name: Build

--- a/package.json
+++ b/package.json
@@ -26,7 +26,8 @@
 	"author": "Aviortheking",
 	"license": "MIT",
 	"devDependencies": {
-		"lucide-static": "0.294.0"
+		"lucide-static": "0.294.0",
+		"semver": "^7.5.4"
 	},
 	"peerDependencies": {
 		"astro": ">=2.7.1"

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
 	"name": "lucide-astro",
-	"version": "0.294.0",
+	"version": "0.294.1",
 	"description": "Get your Lucide icons right into your Astro project",
 	"main": "./dist/index.js",
 	"types": "./dist/index.d.ts",

--- a/upgrade.js
+++ b/upgrade.js
@@ -1,4 +1,5 @@
 import fs from 'node:fs/promises'
+import semver from 'semver'
 /**
  * Javascript script linked to `upgrade.sh`
  */
@@ -10,6 +11,7 @@ const upstreamPackage = 'lucide-static'
 
 	const check = await fetch(`https://registry.npmjs.com/${upstreamPackage}`).then((it) => it.json())
 	const newVersion = check['dist-tags']?.latest
+
 	if (!newVersion) {
 		console.error('no new version found :(')
 		process.exit(1)
@@ -18,6 +20,12 @@ const upstreamPackage = 'lucide-static'
 		console.log('no diff in version, ending')
 		process.exit(1)
 	}
+
+	if (!semver.gt(newVersion, pkgVersion)) {
+		console.log('package version', pkgVersion, 'higher than upstream version', newVersion, 'skipping update')
+		process.exit(1)
+	}
+
 	console.log('diff in version, update needed')
 	pkg.version = newVersion
 	pkg.devDependencies[upstreamPackage] = newVersion

